sw/device/lib/base/bitfield.h

```diff
@@ -274,6 +274,18 @@ inline uint32_t bitfield_byteswap32(uint32_t bitfield) {
   return __builtin_bswap32(bitfield);
 }
 
+/**
+ * Check whether the bitfield value is power of two algned.
+ *
+ * Zero will also return false.
+ *
+ * @param bitfield Value to be verified.
+ * @return True if bitfield is power of two, otherwise false.
+ */
+inline bool bitfield_is_power_of_two32(uint32_t bitfield) {
+  return bitfield && !(bitfield & (bitfield - 1));
```

Comment:
Can this avoid using implicit conversions? I.e., use `bitfield != 0 && (bitfield & (bitfield - 1)) == 0`. Or even

```c
if (bitfield == 0) { return false; }
return (bitfield & (bitfield - 1)) == 0;
```

(LLVM will remove the branch, no need to worry.)

Reply:
done